src/Polly.Core/Utils/TaskHelper.cs

```diff
@@ -1,7 +1,40 @@
 ﻿namespace Polly.Utils;
 
+#pragma warning disable S5034 // "ValueTask" should be consumed correctly
+
 internal static class TaskHelper
 {
+    public static void GetResult(this ValueTask<VoidResult> task)
+    {
+        Debug.Assert(
+            task.IsCompleted,
+            "The value task should be already completed at this point. If not, it's an indication that the strategy does not respect the ResilienceContext.IsSynchronous value.");
```

Comment:
Does this message need updating?

Reply:
I think here it's ok. Component does not care about sync vs async. It's only `ResilienceStrategy` that won't respect this flag.